allennlp/commands/elmo.py

```diff
@@ -0,0 +1,203 @@
+The output is a hdf5 file (http://docs.h5py.org/en/latest/) where each
+sentence is a size (3, num_tokens, 1024) array with the biLM representations.
+
+In the default setting, each sentence is keyed in the output file by the line number
+in the original text file.  Optionally, by specifying --use_sentence_key
+the first token in each sentence is assumed to be a unique sentence key
+used in the output file.
+"""
+
+import logging
+from typing import IO, List, Iterable, Tuple
+
+import argparse
+import h5py
+import torch
+
+from allennlp.common.tqdm import Tqdm
+from allennlp.common.util import lazy_groups_of
+from allennlp.data.dataset import Batch
+from allennlp.data import Token, Vocabulary, Instance
+from allennlp.data.fields import TextField
+from allennlp.data.token_indexers.elmo_indexer import ELMoTokenCharactersIndexer
+from allennlp.nn.util import remove_sentence_boundaries
+from allennlp.modules.elmo import _ElmoBiLm
+from allennlp.commands.subcommand import Subcommand
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+DEFAULT_OPTIONS_FILE = "https://s3-us-west-2.amazonaws.com/allennlp/models/elmo/2x4096_512_2048cnn_2xhighway/elmo_2x4096_512_2048cnn_2xhighway_options.json" # pylint: disable=line-too-long
+DEFAULT_WEIGHT_FILE = "https://s3-us-west-2.amazonaws.com/allennlp/models/elmo/2x4096_512_2048cnn_2xhighway/elmo_2x4096_512_2048cnn_2xhighway_weights.hdf5" # pylint: disable=line-too-long
+
+
+class Elmo(Subcommand):
+    def add_subparser(self, name: str, parser: argparse._SubParsersAction) -> argparse.ArgumentParser:
+        # pylint: disable=protected-access
+        description = '''Create word vectors using ELMo.'''
+        subparser = parser.add_parser(
+                name, description=description, help='Use a trained model to make predictions.')
+
+        subparser.add_argument('input_file', type=argparse.FileType('r'), help='path to input file')
+        subparser.add_argument('output_file', type=str, help='path to output file')
+
+        subparser.add_argument('--vocab-path', type=str, help='A path to a vocabulary file to generate ')
+        subparser.add_argument(
+                '--options-file',
+                type=str,
+                default=DEFAULT_OPTIONS_FILE,
+                help='The path to the ELMo options file.')
+        subparser.add_argument(
+                '--weight-file',
+                type=str,
+                default=DEFAULT_WEIGHT_FILE,
+                help='The path to the ELMo weight file.')
+        subparser.add_argument('--batch-size', type=int, default=64, help='The batch size to use.')
+        subparser.add_argument('--cuda-device', type=int, default=-1, help='The cuda_device to run on.')
+        subparser.add_argument('--use-sentence-key', default=False, action='store_true')
+
+        subparser.set_defaults(func=elmo_command)
+
+        return subparser
+
+
+class ElmoEmbedder():
```

Comment:
Our convention seems to be one class per file.  Where should I move this?

Reply:
do we expect that anyone besides this command would use this class? if so, put it where they'd "expect" it to be. if not, IMO it's fine to leave it here (and maybe make it "private")

Reply:
This class would be the easiest way for someone to get ELMo vectors for sentences programatically--so yes, I think I do.